### PR TITLE
Add unit tests and malformed stream detection for parseSseStream

### LINKED ISSUE #13: Add unit tests for parseSseStream helper
## Background

`apps/frontend/src/api/helpers/sse.ts` contains `parseSseStream`, which handles SSE stream parsing with buffering across chunk boundaries. This is a pure logic function with no UI dependencies, making it a good candidate for unit tests.

## Test cases

- Single event in one chunk
- Multiple events in one chunk
- Event split across multiple chunks (partial buffer)
- Empty lines and non-`data:` lines are skipped
- Empty response body throws error

## Location

`apps/frontend/src/api/helpers/sse.test.ts`

### PR DESCRIPTION
**Pull request by Soulike**
## Summary
- Add comprehensive unit tests for `parseSseStream` covering happy paths, buffering edge cases, filtering, error handling, and large-scale scenarios
- Detect malformed SSE events that don't match any valid field prefix (`data:`, `event:`, `id:`, `retry:`, `:`) and throw an error to surface server-side bugs early
- 44 tests covering single/multiple events, chunk boundary buffering, valid SSE field skipping, malformed stream detection, and large payloads (100K chars, 1000 events)

Closes #13

## Test plan
- [x] All 44 unit tests pass (`bun run test`)
- [x] ESLint passes on both changed files
- [x] Pre-commit hooks (prettier + eslint) pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/frontend/src/api/helpers/sse.ts [hunk 1]
```diff
@@ -38,7 +48,13 @@ export async function* parseSseStream(
 
     for (const part of parts) {
       const trimmed = part.trim();
-      if (trimmed === '' || !trimmed.startsWith(SSE_DATA_PREFIX)) continue;
+      if (trimmed === '') continue;
+
+      if (!isSseFieldValid(trimmed)) {
+        throw new Error(`Malformed SSE event: ${trimmed}`);
+      }
```
[the remark this answers is not in the document]
Comment (Soulike) [reply apps/frontend/src/api/helpers/sse.ts]: Fixed: Error messages now truncate malformed content to 120 characters with a char count suffix for longer payloads.

File: apps/frontend/src/api/helpers/sse.test.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+  describe('non-data lines', () => {
+    it('should skip event blocks without data: prefix', async () => {
+      const response = createMockResponse([
+        'event: message\n\ndata: hello\n\n',
+      ]);
+      const results = await collectResults(response);
+      expect(results).toEqual(['hello']);
+    });
+
```
[the remark this answers is not in the document]
Comment (Soulike) [reply apps/frontend/src/api/helpers/sse.test.ts]: Fixed: Added test for multi-field event (event: message + data: hello in one block).

File: apps/frontend/src/api/helpers/sse.test.ts [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+      const results = await collectResults(response);
+      expect(results).toEqual([json]);
+    });
+
+    it('should preserve spaces in data content', async () => {
+      const response = createMockResponse(['data: hello world\n\n']);
+      const results = await collectResults(response);
+      expect(results).toEqual(['hello world']);
+    });
+
```
[the remark this answers is not in the document]
Comment (Soulike) [reply apps/frontend/src/api/helpers/sse.test.ts]: Fixed: Added test asserting trailing spaces in data content are preserved.

File: apps/frontend/src/api/helpers/sse.ts [hunk 4]
```diff
@@ -22,25 +86,33 @@ export async function* parseSseStream(
...
-      if (trimmed === '' || !trimmed.startsWith(SSE_DATA_PREFIX)) continue;
+        const data = extractDataFromBlock(part);
+        if (data === undefined) continue;
 
-      yield trimmed.slice(SSE_DATA_PREFIX.length);
+        yield data;
+      }
     }
+  } finally {
+    lineReader.releaseLock();
```
[the remark this answers is not in the document]
Comment (Soulike) [reply apps/frontend/src/api/helpers/sse.ts]: Fixed: Replaced releaseLock() with cancel() to properly abort the underlying fetch connection and free resources on early exit or error.

File: apps/frontend/src/api/helpers/sse.ts [hunk 5]
```diff
@@ -22,25 +91,33 @@ export async function* parseSseStream(
...
-      if (trimmed === '' || !trimmed.startsWith(SSE_DATA_PREFIX)) continue;
+        const data = extractDataFromBlock(part);
+        if (data === undefined) continue;
 
-      yield trimmed.slice(SSE_DATA_PREFIX.length);
+        yield data;
+      }
     }
+  } finally {
+    await lineReader.cancel();
```
[the remark this answers is not in the document]
Comment (Soulike) [reply apps/frontend/src/api/helpers/sse.ts]: Fixed: Wrapped cancel() in .catch() so a cancel failure in finally doesn't mask the original parsing error.
